packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -654,62 +683,27 @@ export function onForkTransitive<T extends ISubscribable<{ fork: (t: T) => void
 }
 
 class RevertibleRevision implements Revertible {
```

Comment:
This class doesn't seem useful anymore. This could all trivially be inlined at the place you generate them. Thoughts?

Reply:
I mainly kept it so that the status api would be called as a getter rather than a function. I just feel like it's a bit nicer but I'm cool with getting rid of the class if you think that's better

Reply:
[You can put getters on object properties.](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhjAvDA3gKBpmBzAprAIwAoBKALhjAFcBbA3AJ1RgfyobBgGYYBfNfmjgA6AgG4gA)

Reply:
I'd prefer to get rid of it, as the class is entirely boilerplate. I like Noah's suggestion.

Reply:
omg that changes everything, I asked chatgpt if I could and it basically told me no so that's very annoying

Reply:
"We will lose our jobs because of AI. Not because it will replace us, but because it will make us into bad developers." - Franklin D. Roosevelt, according to Microsoft Copilot